src/server/session.ts

```diff
@@ -463,17 +466,41 @@ module ts.server {
             var edits = compilerService.languageService.getFormattingEditsAfterKeystroke(file, position, key,
                 compilerService.formatCodeOptions); 
             if ((key == "\n") && ((!edits) || (edits.length == 0) || allEditsBeforePos(edits, position))) {
-                // TODO: get these options from host
-                var editorOptions: ts.EditorOptions = {
-                    IndentSize: 4,
-                    TabSize: 4,
-                    NewLineCharacter: "\n",
-                    ConvertTabsToSpaces: true,
-                };
-                var indentPosition = compilerService.languageService.getIndentationAtPosition(file, position, editorOptions);
-                var spaces = generateSpaces(indentPosition);
-                if (indentPosition > 0) {
-                    edits.push({ span: ts.createTextSpanFromBounds(position, position), newText: spaces });
+                var scriptInfo = compilerService.host.getScriptInfo(file);
+                if (scriptInfo) {
+                    var lineInfo = scriptInfo.getLineInfo(line);
+                    if (lineInfo && (lineInfo.leaf) && (lineInfo.leaf.text)) {

```

Comment:
Why do you only need to do the below if you have a leaf for this line?  What would it mean if you didn't?